component/src/test/java/org/wso2/extension/siddhi/execution/ml/classification/perceptron/PerceptronClassifierStreamProcessorExtensionTestCase.java

```diff
@@ -0,0 +1,517 @@
+                    "streamingml:perceptronClassifier"));
+        }
+    }
+
+    @Test
+    public void testClassificationStreamProcessorExtension8() throws InterruptedException {
+        logger.info("PerceptronClassifierUpdaterStreamProcessorExtension TestCase - default threshold");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream StreamA (attribute_0 double, attribute_1 double, attribute_2 " +
+                "double, attribute_3 double);";
+        String query = ("@info(name = 'query1') from StreamA#streamingml:perceptronClassifier('model1', " + "0.0, " +
+                "attribute_0, attribute_1, attribute_2, attribute_3) \n" + "insert all events into outputStream;");
+
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(trainingStream + inStreamDefinition
+                + trainingQuery + query);
+        siddhiAppRuntime.addCallback("query1", new QueryCallback() {
+
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                count++;
+                EventPrinter.print(inEvents);
+                // should emit probabilities
+                if (count == 1) {
+                    AssertJUnit.assertArrayEquals(new Object[]{0.8, 0.67, 0.1, 0.03, true, 0.5263760000000001}, inEvents[0]
+                            .getData());
+                } else if (count == 2) {
+                    AssertJUnit.assertArrayEquals(new Object[]{0.33, 0.23, 0.632, 0.992, false, 0.2779144}, inEvents[0]
+                            .getData());
+                }
+            }
+        });
+        try {
+            InputHandler inputHandler = siddhiAppRuntime.getInputHandler("StreamTrain");
+            siddhiAppRuntime.start();
+            inputHandler.send(new Object[]{1.0, 1.0, 0.2, 0.13, "true"});
+            inputHandler.send(new Object[]{0.9, 0.89, 0.3, 0.02, "true"});
+            inputHandler.send(new Object[]{0.0, 0.0, 1.0, 0.82, "false"});
+            inputHandler.send(new Object[]{0.01, 0.4, 0.77, 0.92, "false"});
+            inputHandler.send(new Object[]{0.80, 0.81, 0.11, 0.13, "true"});
+            inputHandler.send(new Object[]{0.02, 0.30, 0.88, 0.76, "false"});
+            inputHandler.send(new Object[]{0.93, 0.71, 0.02, 0.122, "true"});
+            inputHandler.send(new Object[]{0.29, 0.24, 0.98, 0.65, "false"});
+            Thread.sleep(5000);
+
+            InputHandler inputHandler1 = siddhiAppRuntime.getInputHandler("StreamA");
+            // send some unseen data for prediction
+            inputHandler1.send(new Object[]{0.8, 0.67, 0.1, 0.03});
+            inputHandler1.send(new Object[]{0.33, 0.23, 0.632, 0.992});
+            AssertJUnit.assertEquals(2, count);
+            Thread.sleep(1100);
+        } catch (Exception e) {
+            logger.error(e.getMessage());
```

Comment:
What is the reason to catch the exception. Shouldn't the testcase fail in case of an exception. Applies to other testcases as well

Reply:
Updated the testcase in [#9ff2603421633442dc5c5ef0edde7f70a8c2a1a7](https://github.com/wso2-extensions/siddhi-execution-streamingml/pull/48/commits/9ff2603421633442dc5c5ef0edde7f70a8c2a1a7)